src/imoinfo.h

```diff
@@ -294,53 +295,59 @@ class QDLLEXPORT IOrganizer: public QObject {
+  virtual void refreshModList(bool saveChanges = true) = 0;
 
   /**
-   * @param func function to be called when an application is run
+   * @brief get the currently managed game info
    */
-  virtual bool onAboutToRun(const std::function<bool(const QString&)> &func) = 0;
+  virtual MOBase::IPluginGame const *managedGame() const = 0;
 
   /**
-   * @param func function to be called when an application is run
+   * @brief Get the mod list, sorted by current profile priority
    */
-  virtual bool onFinishedRun(const std::function<void(const QString&, unsigned int)> &func) = 0;
+  virtual QStringList modsSortedByProfilePriority() const = 0;
 
   /**
-   * @brief refresh the mod list
-   * @param saveChanges if true, the relevant profile information is saved first (enabled mods and the ordering)
+   * @param func Function to called when a mod has been installed. The parameter to the callback is the mod name.
    */
-  virtual void refreshModList(bool saveChanges = true) = 0;
+  virtual bool onModInstalled(const std::function<void(const QString&)>& func) = 0;
 
   /**
-   * @brief get the currently managed game info
+   * @param func Function to be called when an application is run.
    */
-  virtual MOBase::IPluginGame const *managedGame() const = 0;
+  virtual bool onAboutToRun(const std::function<bool(const QString&)>& func) = 0;
 
   /**
-   * @brief Get the mod list, sorted by current profile priority
+   * @param func Function to be called when an application is run.
    */
-  virtual QStringList modsSortedByProfilePriority() const = 0;
+  virtual bool onFinishedRun(const std::function<void(const QString&, unsigned int)>& func) = 0;
 
-Q_SIGNALS:
+  /**
+   * @param func Function to be called when the user interface has been initialized.
+   */
+  virtual bool onUserInterfaceInitialized(std::function<void(QMainWindow*)> const& func) = 0;
 
   /**
-   * Signal emitted when a setting for a plugin change.
+   * @param func Function to be called when the current profile change. The first argument is the old
+   *   profile, which can be null, and the second one the new profile (cannot be null).
    *
-   * @param pluginName Name of the plugin.
-   * @param key Name of the setting.
-   * @param oldValue Old value of the setting. Can be a default-constructed QVariant if the setting did not
+   */
+  virtual bool onProfileChanged(std::function<void(IProfile*, IProfile*)> const& func) = 0;
+
+  /**
+   * @param func Function to be called when a plugin setting is changed. 
+   *
+   * @fparam pluginName Name of the plugin.
+   * @fparam key Name of the setting.
+   * @fparam oldValue Old value of the setting. Can be a default-constructed QVariant if the setting did not
    *   exist before.
-   * @param newValue New value of the setting. Can be a default-constructed QVariant if the setting has been
+   * @fparam newValue New value of the setting. Can be a default-constructed QVariant if the setting has been

```

Comment:
`fparam`?

Reply:
It's a parameter of the callback, I did not really know how to put it... I did not want to put everything in the description of `func` but since there names are not visible in the `std::function<>`, this does not actually make sense...

Reply:
Modified in ab5124b